packages/admin-ui/src/filters/single-filter.state.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react'
+import type { FilterItem, key, UseFilterStateReturn } from './filter.state'
+import { useFilterState } from './filter.state'
+
+export function useSingleFilterState<T extends FilterItem>(
+  props: UseSingleFilterStateProps<T>
+): UseSingleFilterReturn<T> {
+  const { initialApplied, onChange: onChangeCb, ...otherProps } = props
+
+  const onChange = ({ selected }: { selected: key[] }) => {
+    onChangeCb({ selected: selected?.length ? selected[0] : null })
+  }
+
+  const filterState = useFilterState({
+    ...otherProps,
+    initialApplied: initialApplied ? [initialApplied] : [],
+    onChange,
+    selectionMode: 'single',
+  })
+
+  const { appliedValues, appliedKeys, selectedKeys, ...singleSelectState } =
+    filterState
+
+  // forces apply when one item is selected
+  useEffect(() => {
+    filterState.onChange()
+  }, [selectedKeys[0]])
```

Comment:
Why force if it has the apply button? 

Every time I select an option the popover hides and I think it can lead to a bad UX, mainly if you are navigating by the keyboard. I think is something to align with davis or peu

Reply:
We've discussed this issue on slack, so i'm resolving the conversation